test/src/main/java/com/rakuten/test/MainApplication.kt

```diff
@@ -9,9 +12,35 @@ class MainApplication : Application() {
     lateinit var settings: IAMSettings
 
     override fun onCreate() {
+        setupStrictMode()
         super.onCreate()
         settings = IAMSettings(this)
         InAppMessaging.configure(this)
         InAppMessaging.instance().registerPreference(provider)
     }
+
+    /**
+     * StrictMode is most commonly used to catch accidental disk or network access on the application's main thread,
+     * where UI operations are received and animations take place.
+     * https://developer.android.com/reference/android/os/StrictMode
+     */
+    private fun setupStrictMode() {
+        StrictMode.setThreadPolicy(
+            ThreadPolicy.Builder()
+                //.detectDiskReads()
+                //.detectDiskWrites()
+                .detectNetwork() // or .detectAll() for all detectable problems
+                .penaltyLog()
+                .build()
+        )
+        // VM Policy - focused on memory leaks
+        StrictMode.setVmPolicy(
```

Comment:
@maureenorea-clores 
Please add `detectNonSdkApiUsage()` to detect reflective usage of non-Android SDK APIs.
nit: keep detect Disk reads/writes for UI thread usage violations.